resources/python/src/nuSQUIDSpy.h

```diff
@@ -92,6 +92,11 @@ struct marray_from_python{
     //accept only numpy arrays
     if(!PyArray_Check(obj_ptr))
       return(NULL);
+    // Analogously to what is described in
+    // https://docs.python.org/3/c-api/intro.html#reference-counts,
+    // the call below always increases the reference count of the object by one and we 
+    // are left with the responsibility to decrease the reference count when we are done
+    // with it.
     PyArrayObject* numpy_array=PyArray_GETCONTIGUOUS((PyArrayObject*)obj_ptr);
```

Comment:
It seems like `PyArray_GETCONTIGUOUS ` may not actually be needed at all; if we can replace it with a simple cast which neither allocates a new array nor increases the reference count of the old one this whole concerns would go away (for this function).

Reply:
Maybe we could, but I don't feel confident enough about it to do unorthodox things here. As `PyArray_GETCONTIGUOUS` also just returns a reference if the source is already contiguous, I don't think that we are losing any performance in this case.